packages/rum-recorder/src/domain/rrweb/mutation.spec.ts

```diff
@@ -101,3 +109,40 @@ function createNodeList(nodes: Node[]): NodeList {
     },
   }) as unknown) as NodeList
 }
+
+class MockMutationObserver implements MutationObserver {
+  static instances: MockMutationObserver[] = []
```

Comment:
when do we need to handle multiple instances?

Reply:
No we don't have to. Do you see any issue with this?

Reply:
what about simplifying the implementation to only deal with a single instance?
it could ease understanding / maintenance